src/routes/github/github-router.ts

```diff
@@ -27,6 +31,14 @@ subRouter.post("/webhooks",
 	returnOnValidationError,
 	WebhookReceiverPost);
 
+// We want to restrict the tail of OAuth flow to the same scope as the starting point had (where GitHubOAuthMiddleware
+// fired), therefore we are not including neither jira*middlewares nor github*middlewares. GithubOAuthCallbackGet will
+// get all the data needed from session to obtain the token and then redirect to the original URL with all the
+// restrictions that are in place there.
+// We don't want to artificially limit ourselves by including those middlewares, because there are scenarios when
+// OAuth flow was triggered outside of Jira admin scope (e.g. create-branch, or approve-connection).
+subRouter.use(OAUTH_CALLBACK_SUBPATH, GithubOAuthCallbackGet);
```

Comment:
We can move callback router to after the `jiraSymmetricJwtMiddleware` and `GithubServerAppMiddleware` but before `jiraAdminPermissionsMiddleware`,  so that still allow non jira-admin user for callback.

Reply:
This puts more limitation, to have jiraHost in session (cause otherwise it will be rejected). The callback is just finishing the flow that has been started and the state is passed over, therefore IMHO it is not needed and may prohibit us doing certain things (e.g. when authorization is granted not based on JWT or jiraHost but some random token.

Reply:
Okay I know what you are doing. In the code, I find it hard to co-relate.

In the callback, it has following code.
```
const jiraHost = (await Installation.findByPk(state.installationIdPk))?.jiraHost;
```

For me, this can be a **security risk**, as the state contains info for important to security. The code in PR will **work**, but it depends on we have to carefully on constructing the oauth redirect url duirng loginGet, we need to verify user is a jiraAdmin before we put that thing in state. I mean the security of the `Callback` (outside of the `jiraSymmetricJwtMiddleware` ) relies on the `GithubOAuthLoginGet` (being protected by `jiraSymmetricJwtMiddleware`). <-- this cause the confusing in understanding security wise on these middlewares.

Further more, I think we could be using something wrong in the state from this link:
https://auth0.com/docs/secure/attack-protection/state-parameters#:~:text=The%20primary%20reason%20for%20using,matches%20the%20one%20you%20sent.

```
The primary reason for using the state parameter is to mitigate CSRF attacks by using a unique and non-guessable value associated with each authentication request about to be initiated. That value allows you to prevent the attack by confirming that the value coming from the response matches the one you sent.
```

Reply:
But again, let's chat about the big picture of this "Approval flow". It will make more sense once we understand it.